src/vcpkg/postbuildlint.cpp

```diff
@@ -1281,13 +1282,17 @@ namespace vcpkg
         const auto stringview_paths = Util::fmap(string_paths, [](std::string& s) { return StringView(s); });
 
         std::vector<Path> failing_files;
-        for (auto&& file : fs.get_regular_files_recursive(dir, IgnoreErrors{}))
-        {
+        std::mutex mtx;
+        auto files = fs.get_regular_files_recursive(dir, IgnoreErrors{});
+        auto work = [&](auto&& file) {
```

Comment:
I get a compiler error (VS 22 17.7.4) when I replace `auto&&` with `path&&`: `cannot convert argument 1 from Path to Path&&`
 It works on GCC. Is this a STL bug?

Reply:
have you tried `const Path&&` ?

Reply:
Doesn't work either. It works with `const Path&` but that's not what I want. The lambda compiles to `vcpkg::check_no_absolute_paths_in'::2'::<lambda_2>::operator()<vcpkg::Path &>`

Reply:
> Unlike the rest of the parallel algorithms, for_each is not allowed to make copies of the elements in the sequence even if they are [TriviallyCopyable](https://en.cppreference.com/w/cpp/named_req/TriviallyCopyable).

From https://en.cppreference.com/w/cpp/algorithm/for_each